### PR TITLE
JaxRsClientFactory: make the Class<Feature> methods actually invokable

### PR DESCRIPTION
**Pull request by stevenschlansker**
If you have a Feature subclass F, that is actually a Class<F> not a
Class<Feature> but it is also a Class<? extends Feature>

### CODE REVIEW

**Comment by stevenschlansker** [comment]
This is technically a source-incompatible change but the old method signatures were nearly useless so it can't really hurt.

**Comment by stevenschlansker** [comment]
Travis failure due to Maven version mismatch still.  Sigh.

File: client/src/main/java/com/opentable/jaxrs/JaxRsClientFactory.java [hunk 1]
```diff
@@ -144,21 +144,19 @@ public synchronized JaxRsClientFactory addFeatureToGroup(JaxRsFeatureGroup group
...
     @SafeVarargs
-    public final synchronized JaxRsClientFactory addFeatureToGroup(JaxRsFeatureGroup group, Class<Feature>... features) {
+    public final synchronized JaxRsClientFactory addFeatureToGroup(JaxRsFeatureGroup group, Class<? extends Feature>... features) {
         Preconditions.checkState(!started, "Already started building clients");
         classFeatureMap.putAll(group, Arrays.asList(features));
 
-        for (Class<Feature> f : features) {
-            LOG.trace("Group {} registers feature {}", group, f);
-        }
+        LOG.trace("Group {} registers feature {}", group, features);
```
Comment (pennello) [inline client/src/main/java/com/opentable/jaxrs/JaxRsClientFactory.java]: "registers feature**s**", maybe?

File: client/src/main/java/com/opentable/jaxrs/DataUriFeature.java [hunk 2]
```diff
@@ -22,9 +22,15 @@ public boolean configure(FeatureContext context) {
...
-                requestContext.abortWith(Response.ok(uri.getSchemeSpecificPart()).build());
+            final URI uri = requestContext.getUri();
+            final String scheme = uri.getScheme();
+            if (scheme.startsWith("data")) {
+                int code = 200;
+                final int plusIdx = scheme.indexOf('+');
+                if (plusIdx > 0) {
+                    code = Integer.parseInt(scheme.substring(plusIdx, scheme.length()));
+                }
+                requestContext.abortWith(Response.status(code).entity(uri.getSchemeSpecificPart()).build());
```
Comment (pennello) [inline client/src/main/java/com/opentable/jaxrs/DataUriFeature.java]: Cute.  Is this according to any sort of standard?

File: client/src/main/java/com/opentable/jaxrs/JaxRsClientFactory.java [hunk 3]
```diff
@@ -183,8 +183,13 @@ public synchronized ClientBuilder newBuilder(String clientName, Collection<JaxRs
...
             .collect(Collectors.toList());
 
-        LOG.debug("Building client '{}' with feature groups {}, features {}, and config '{}'", clientName, featureGroups, features, jaxRsConfig);
+        final List<Class<? extends Feature>> classFeatures = featureGroups.stream()
+            .flatMap(g -> classFeatureMap.get(g).stream())
+            .collect(Collectors.toList());
+
+        LOG.debug("Building client '{}' with feature groups {}, features {} {}, and config '{}'", clientName, featureGroups, features, classFeatures, jaxRsConfig);
         features.forEach(builder::register);
+        classFeatures.forEach(builder::register);
```
Comment (pennello) [inline client/src/main/java/com/opentable/jaxrs/JaxRsClientFactory.java]: Ah, we were just missing these before.

File: client/src/main/java/com/opentable/jaxrs/DataUriFeature.java (hunk 2 above)
Comment (stevenschlansker) [inline client/src/main/java/com/opentable/jaxrs/DataUriFeature.java]: Not that I'm aware of

Comment (pennello) [inline client/src/main/java/com/opentable/jaxrs/DataUriFeature.java]: All right.  It's cool anyway.

**Pull request merged**